src/core/ddsi/src/ddsi_raweth.c

```diff
@@ -210,6 +302,34 @@ static dds_return_t ddsi_raweth_create_conn (struct ddsi_tran_conn **conn_out, s
     return DDS_RETCODE_ERROR;
   }
 
+  memset(&mreq, 0, sizeof (mreq));
+  mreq.mr_ifindex = (int)intf->if_index;
+  mreq.mr_type = PACKET_MR_PROMISC;
```

Comment:
It is somewhat surprising to me that promiscuous mode is needed, or is that because of the `ETH_P_ALL`?

Reply:
It appears that promiscuous mode is necessary because the linux kernel will drop frames with a vlan tag for which no vlan is configured.